.github/workflows/operator-e2e-integration-tests.yml

```diff
@@ -67,5 +67,45 @@ jobs:
         run: |
           # Delete the KIND cluster after tests
           kind delete cluster --name kind-$KIND_CLUSTER
+          docker images --format '{{.Repository}}:{{.Tag}}' | grep 'feast' | xargs -r docker rmi -f
+          docker system prune -a -f
 
-      
+      - name: Run Previous version tests
+        run: |
+          # Run the previous version tests
+          cd infra/feast-operator/
+          make test-previous-version
+
+      - name: Debug KIND Cluster when there is a failure
+        if: failure()
+        run: |
+          kubectl get pods --all-namespaces
+          kubectl describe nodes
+
+      - name: Clean up
+        if: always()
+        run: |
+          # Delete the KIND cluster after tests
+          kind delete cluster --name kind-$KIND_CLUSTER
+          docker images --format '{{.Repository}}:{{.Tag}}' | grep 'feast' | xargs -r docker rmi -f
+          docker system prune -a -f
+
+      - name: Run Upgrade tests
+        run: |
+          # Run the upgrade tests
+          cd infra/feast-operator/
+          make test-upgrade
+
+      - name: Debug KIND Cluster when there is a failure
+        if: failure()
+        run: |
+          kubectl get pods --all-namespaces
+          kubectl describe nodes
+
+      - name: Clean up
+        if: always()
```

Comment:
why do we need clean up multiple times? deleting the kind cluster in previous clean up and is the test expected to run without kind cluster or we recreating it?

Reply:
I clean between each step because in each step an image is being created and loaded